test/box/net.box_closing_without_lost_gh-6338.test.lua

```diff
@@ -0,0 +1,164 @@
+-- This test checks streams iplementation in iproto (gh-5860).
+net_box = require('net.box')
+fiber = require('fiber')
+test_run = require('test_run').new()
+
+test_run:cmd("create server test with script='box/iproto_streams.lua'")
```

Comment:
Do you really need to start a new server here? Can't you run the test on self (`net.connect(box.cfg.listen)`)?

Reply:
For one of test cases we need to drop connection on server side and then check result! For this purpose, I use `test_run:cmd("stop server test")` in fiber.

Reply:
Why? AFAIU all you need to check is that
- net.box.close() flushes the send buffer
- new requests will fail if net.box.close() is in progress.

Please simplify the test to test exactly this.

Reply:
Before i add `on_send_buf_empty:broadcast()` in `error_sm` `conn:close` hang. I add this part of test especially for this case.